dpppt-backend-sdk/dpppt-backend-sdk-data/src/main/java/org/dpppt/backend/sdk/data/gaen/JDBCGAENDataServiceImpl.java

```diff
@@ -125,6 +125,53 @@ public void upsertExposeesDelayed(
     return jt.query(sql, params, new GaenKeyRowMapper());
   }
 
+  @Override
+  @Transactional(readOnly = true)
+  public List<GaenKey> getSortedExposedSince(UTCInstant keysSince, UTCInstant now) {
+    MapSqlParameterSource params = new MapSqlParameterSource();
+    params.addValue("since", keysSince.getDate());
+    params.addValue("maxBucket", now.roundToBucketStart(releaseBucketDuration).getDate());
+    params.addValue("timeSkewSeconds", timeSkew.toSeconds());
+
+    // Select keys since the given date. We need to make sure, only keys are returned
+    // that are allowed to be published.
+    // For this, we calculate the expiry for each key in a sub query. The expiry is then used for
+    // the where clause:
+    // - if expiry <= received_at: the key was ready to publish when we received it. Release this
+    // key, if received_at in [since, maxBucket)
+    // - if expiry > received_at: we have to wait until expiry till we can release this key. This
+    // means we only release the key if expiry in [since, maxBucket)
+    // This problem arises, because we only want key with received_at after since, but we need to
+    // ensure that we relase ALL keys meaning keys which were still valid when they were received
+
+    // we need to add the time skew to calculate the expiry timestamp of a key:
+    // TO_TIMESTAMP((rolling_start_number + rolling_period) * 10 * 60 + :timeSkewSeconds
+
+    String sql =
+        "select keys.pk_exposed_id, keys.key, keys.rolling_start_number, keys.rolling_period,"
+            + " keys.transmission_risk_level from (select pk_exposed_id, key,"
+            + " rolling_start_number, rolling_period, transmission_risk_level, received_at,  "
+            + getSQLExpressionForExpiry()
+            + " as expiry from t_gaen_exposed)"
+            + " as keys where ((keys.received_at >= :since AND"
+            + " keys.received_at < :maxBucket AND keys.expiry <= keys.received_at) OR (keys.expiry"
+            + " >= :since AND keys.expiry < :maxBucket AND keys.expiry > keys.received_at))";
```

Comment:
Niceify sql expression to fit with comment.

```suggestion
            + " as keys where ( "(keys.expiry <= keys.received_at AND "
            + " keys.received_at >= :since AND keys.received_at < :maxBucket) OR "
            + "(keys.expiry > keys.received_at AND keys.expiry >= :since AND keys.expiry < :maxBucket) )";
```